kernel/src/main/java/org/kframework/compile/GenerateSentencesFromConfigDecl.java

```diff
@@ -24,620 +33,806 @@
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
-                throw KEMException.compilerError("Malformed external cell in configuration declaration.", term);
-            } else if (KLabels.CELLS.equals(kapp.klabel())) {
-                //is a cell bag, and thus represents the multiple children of its parent cell
-                if (ensures != null) {
-                    //top level cell, therefore, should be the children of the generatedTop cell
-                    KToken cellLabel = KToken(KLabels.GENERATED_TOP_CELL_NAME, Sort("#CellName"));
-                    K generatedTop = KApply(KLabel("#configCell"), cellLabel, KApply(KLabel("#cellPropertyListTerminator")), term, cellLabel);
-                    return genInternal(generatedTop, ensures, cfgAtt, m);
-                }
-                List<K> cells = Assoc.flatten(kapp.klabel(), kapp.klist().items(), m);
-                Set<Sentence> accumSentences = Set();
-                List<Sort> sorts = Lists.newArrayList();
-                List<K> initializers = Lists.newArrayList();
-                for (K cell : cells) {
-                    //for each cell, generate the child and inform the parent of the children it contains
-                    Tuple4<Set<Sentence>, List<Sort>, K, Boolean> childResult = genInternal(cell, null, cfgAtt, m);
-                    accumSentences = (Set<Sentence>)accumSentences.$bar(childResult._1());
-                    sorts.addAll(childResult._2());
-                    initializers.add(childResult._3());
+  /**
+   * Takes a configuration declaration and returns the sentences that it desugars into.
+   *
+   * <p>Cells of multiplicity 1 desugar into an initializer production, an initializer rule, and a
+   * cell production. Cells of multiplicity * desugar into an initializer production, an initializer
+   * rule, a cell production, and a bag sort to represent a bag of those cells. Cells of
+   * multiplicity ? desugar into an initializer production, an initializer rule, a cell production,
+   * and an empty production indicating the absence of that cell. Cells with children additionally
+   * generate a *CellFragment sort with the same arity as the cell production, but the arguments
+   * made optional by generating additional sorts. Cells which have parents and are not multiplicity
+   * * generate a CellOpt sort which is a supersort of the cell sort and has an additional
+   * production name like {@code <cell>-absent}. (For a cell with multiplicitly ? this is necessary
+   * to distinguish a fragment that did capture the state of the cell when it wasn't present, from a
+   * cell fragment that didn't even try to capture the cell).
+   *
+   * <p>Currently the implementation does not handle initializer rules; we will address this
```

Comment:
There was a bit of a structure in this comment. Maybe we can recapture it with `<p>`